util/config.go

```diff
@@ -265,22 +270,27 @@ func getNativeScriptAndroidClasspath(config *Config) string {
 	return path.Join(getNativeScriptAndroidPlatformPath(config), "java", path.Join(strings.Split(config.AndroidManifest.PackageID, ".")...))
 }
 
-func getDefaultAndroidPlatformPath(config *Config) string {
-	return path.Join(config.AppDir, config.AppTarget, "src", "main")
+func getDefaultAndroidPlatformPath(config *Config, useFlavor bool) string {
+	srcPath := path.Join(config.AppDir, config.AppTarget, "src")
+	if (useFlavor && len(config.FlavorName) > 0) {
```

Comment:
I'd prefer to have a check if the dir exists to avoid typos when specifying the flavour name